cpg_workflows/stages/realign_genotype_with_dragen.py

```diff
@@ -0,0 +1,357 @@
+            jobs=monitor_pipeline_run,
+        )
+
+
+@stage(analysis_type='dragen_mlr', required_stages=[AlignGenotypeWithDragen])
+class GvcfMlrWithDragen(SequencingGroupStage):
+    def expected_outputs(
+        self,
+        sequencing_group: SequencingGroup,
+    ) -> None:
+        pass
+
+    def queue_jobs(self, sequencing_group: SequencingGroup, inputs: StageInput) -> StageOutput | None:
+        pass
+
+
+@stage(required_stages=[GvcfMlrWithDragen])
+class MonitorGvcfMlrWithDragen(SequencingGroupStage):
+    def expected_outputs(
+        self,
+        sequencing_group: SequencingGroup,
+    ) -> None:
+        pass
+
+    def queue_jobs(self, sequencing_group: SequencingGroup, inputs: StageInput) -> StageOutput | None:
+        pass
+
+
+@stage(required_stages=[AlignGenotypeWithDragen, GvcfMlrWithDragen])
+class CancelIcaPipelineRun(SequencingGroupStage):
+    def expected_outputs(
+        self,
+        sequencing_group: SequencingGroup,
+    ) -> None:
+        pass
+
+    def queue_jobs(self, sequencing_group: SequencingGroup, inputs: StageInput) -> StageOutput | None:
+        pass
+
+
+@stage(
+    analysis_type='ica_data_download',
+    required_stages=[PrepareIcaForDragenAnalysis, MonitorAlignGenotypeWithDragen, GvcfMlrWithDragen],
+)
+class DownloadDataFromIca(SequencingGroupStage):
+    def expected_outputs(
+        self,
+        sequencing_group: SequencingGroup,
+    ) -> cpg_utils.Path:
+        bucket_name: str = get_path_components_from_gcp_path(path=str(object=sequencing_group.cram))['bucket']
+        return cpg_utils.to_path(f'gs://{bucket_name}/{GCP_FOLDER_FOR_ICA_DOWNLOAD}/{sequencing_group.name}')
+
+    def queue_jobs(self, sequencing_group: SequencingGroup, inputs: StageInput) -> StageOutput | None:
+        cram_path_components: dict[str, str] = get_path_components_from_gcp_path(path=str(object=sequencing_group.cram))
+        cram: str = f'{cram_path_components["suffix"]}{cram_path_components["file"]}'
+        bucket_name: str = cram_path_components['bucket']
+        ica_analysis_folder_id_path: str = str(
+            inputs.as_path(target=sequencing_group, stage=PrepareIcaForDragenAnalysis, key='analysis_output_fid'),
+        )
+        batch_instance: Batch = get_batch()
+        ica_download_job: BashJob = batch_instance.new_bash_job(
+            name='DownloadDataFromIca',
+            attributes=(self.get_job_attrs() or {}) | {'tool': 'ICA'},
+        )
+        ica_download_job.storage(storage=calculate_needed_storage(cram=cram, bucket_name=bucket_name))
+        ica_download_job.image(image=image_path('ica'))
+
+        # Get secrets and folder ID needed for runtime
+        ica_analysis_folder_id = batch_instance.read_input(ica_analysis_folder_id_path)
+
+        # Download an entire folder with ICA. Don't log projectId or API key
+        authenticate_cloud_credentials_in_job(ica_download_job)
+        ica_download_job.command('set +x')
```

Comment:
The performance of this 12-command block would be the same if it was all done as one multiline string block like this [example](https://github.com/populationgenomics/production-pipelines/blob/main/cpg_workflows/jobs/vqsr.py#L558). There's probably as much boilerplate as there is actual  commands so it's n t trivial to see the whole command sequence at a glance

Reply:
Can do that, it's much cleaner